lib/net/dhcp/record.rb

```diff
@@ -64,7 +67,7 @@ def valid?
       self == proxy.record(network, mac)
     end
 
-    def ==(other)
+    def eql_for_conflicts?(other)
```

Comment:
Should we keep the old method? Does something rely on it?

Reply:
It doesn't seems so, but all the tests are testing the old method, we should change the tests to test the new one instead.